plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueCredentialsProvider.java

```diff
@@ -45,10 +48,24 @@ public GlueCredentialsProvider(GlueHiveMetastoreConfig config)
                 provider = DefaultAWSCredentialsProviderChain.getInstance();
             }
             if (config.getIamRole().isPresent()) {
+                AWSSecurityTokenServiceClientBuilder stsClientBuilder = AWSSecurityTokenServiceClientBuilder
+                        .standard()
+                        .withCredentials(provider);
+
+                if (config.getGlueStsEndpointUrl().isPresent() && config.getGlueStsRegion().isPresent()) {
+                    stsClientBuilder.setEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(config.getGlueStsEndpointUrl().get(), config.getGlueStsRegion().get()));
+                }
+                else if (config.getGlueStsRegion().isPresent()) {
+                    stsClientBuilder.setRegion(config.getGlueStsRegion().get());
+                }
+                else if (config.getPinGlueClientToCurrentRegion()) {
+                    stsClientBuilder.setRegion(getCurrentRegionFromEC2Metadata().getName());
+                }
+
                 provider = new STSAssumeRoleSessionCredentialsProvider
                         .Builder(config.getIamRole().get(), "trino-session")
                         .withExternalId(config.getExternalId().orElse(null))
-                        .withLongLivedCredentialsProvider(provider)
```

Comment:
Note for reviewers: `withLongLivedCredentialsProvider` is deprecated and cannot be used along with `withStsClient`, that's why this change is here.